views-core/src/main/java/io/micronaut/views/turbo/TurboStream.java

```diff
@@ -0,0 +1,483 @@
+                    targetCssQuerySelector,
+                    template);
+        }
+
+        /**
+         *
+         * @return The TurboStream template view name.
+         */
+        @NonNull
+        public Optional<String> getTemplateView() {
+            return Optional.ofNullable(templateView);
+        }
+
+        /**
+         *
+         * @return The TurboStream template model.
+         */
+        @NonNull
+        public Optional<Object> getTemplateModel() {
+            return Optional.ofNullable(templateModel);
+        }
+
+        /**
+         * @return TurboFrame Target DOM ID.
+         */
+        @NonNull
+        public Optional<String> getTargetDomId() {
+            return Optional.ofNullable(targetDomId);
+        }
+
+        /**
+         * @return TurboFrame Target DOM ID.
+         */
+        @NonNull
+        public Optional<String> getTargetCssQuerySelector() {
+            return Optional.ofNullable(targetCssQuerySelector);
+        }
+
+        /**
+         * Creates a Turbo Stream builder if annotation {@link TurboView} is found in the route and the request is a turbo request.
+         * @param request HTTP Request
+         * @param response HTTP Response
+         * @return Creates a Turbo Stream builder if annotation {@link TurboView} is found in the route and the request is a turbo request.
+         */
+        @NonNull
+        public static Optional<TurboStream.Builder> of(@NonNull HttpRequest<?> request,
+                                                       @NonNull HttpResponse<?> response) {
+            Optional<AnnotationMetadata> routeMatch = response.getAttribute(HttpAttributes.ROUTE_MATCH,
+                    AnnotationMetadata.class);
+            if (!routeMatch.isPresent()) {
+                return Optional.empty();
+            }
+            return of(request, routeMatch.get());
+        }
+
+        private static Optional<TurboStream.Builder> of(@NonNull HttpRequest<?> request,
+                                                        @NonNull AnnotationMetadata route) {
+            Optional<String> turboFrameOptional = request.getHeaders().get(TurboHttpHeaders.TURBO_FRAME, String.class);
+            if (!route.hasAnnotation(TurboView.class) ||
+                    (!TurboMediaType.acceptsTurboStream(request) && !turboFrameOptional.isPresent() && route.hasAnnotation(View.class))) {
+                return Optional.empty();
+            }
+            TurboStream.Builder builder = TurboStream.builder();
+            route.getValue(TurboView.class, String.class).ifPresent(builder::templateView);
+            route.getValue(TurboView.class, MEMBER_ACTION, TurboStreamAction.class).ifPresent(builder::action);
+            route.getValue(TurboView.class, MEMBER_TARGET_DOM_ID, String.class).ifPresent(builder::targetDomId);
+            route.getValue(TurboView.class, MEMBER_TARGET_CSS_QUERY_SELECTOR, String.class).ifPresent(builder::targetCssQuerySelector);
```

Comment:
using `stringValue(..)` and `enumValue(..)` is more performant

Reply:
`enumValue` does not work. I think we have a bug. I am going to report it in core.